packages/runner/injection/patches/cookies.ts

```diff
@@ -88,26 +95,46 @@ export const patchDocumentCookie = (requestCookies: AutomationCookie[]) => {
       const stringValue = `${newValue}`
       const parsedCookie = CookieJar.parse(stringValue)
 
+      if (parsedCookie?.hostOnly === null) {
+        // we want to make sure the hostOnly property is respected when syncing with CDP/extension to prevent duplicates.
+        // in the case it is not set, we need to calculate it
+        parsedCookie.hostOnly = isHostOnlyCookie(parsedCookie.domain || domain)
+      }
+
       // if result is undefined, it was invalid and couldn't be parsed
       if (!parsedCookie) return getDocumentCookieValue()
 
-      // we should be able to pass in parsedCookie here instead of the string
-      // value, but tough-cookie doesn't recognize it using an instanceof
-      // check and throws an error. because we can't, we have to massage
-      // some of the properties below to be correct
-      const cookie = setCookie(stringValue)
-
-      if (cookie) {
-        cookie.sameSite = parsedCookie.sameSite
-
-        if (!parsedCookie.path) {
-          cookie.path = '/'
+      // if the cookie is not expired, don't set it
```

Comment:
```suggestion
      // if the cookie is not expired, don't remove it
```

Reply:
```suggestion
      // Remove expired cookies
```

Reply:
Changed the comments for what is actually happening in https://github.com/cypress-io/cypress/commit/a67c06d4b42f9ed3d7368bafe6940d4648bec1d0